interp/run.go

```diff
@@ -3087,9 +3087,9 @@ func isNil(n *node) {
 		fnext := getExec(n.fnext)
 		if c0.typ.cat == interfaceT {
 			n.exec = func(f *frame) bltn {
-				vi := value(f).Interface().(valueInterface)
-				if (vi == valueInterface{} ||
-					vi.node.kind == basicLit && vi.node.typ.cat == nilT) {
+				v := value(f)
+				vi, ok := v.Interface().(valueInterface)
+				if ok && (vi == valueInterface{} || vi.node.kind == basicLit && vi.node.typ.cat == nilT) || v.IsNil() {
```

Comment:
Possible optimization: if we start by testing v.IsNil(), then we don't even need to systematically do the type assertion, i.e.:

```
if v.IsNil() {
    // blabla
    // return tnext
}
vi, ok := v.Interface().(valueInterface)
if ok && (vi == valueInterface{} || vi.node.kind == basicLit && vi.node.typ.cat == nilT) {
...
```

However what I'm proposing is not exactly equivalent because in the current version we _only_ do v.isNil() if the type assertion is not ok, and maybe that order is important. So I'm not sure.

Reply:
Your suggestion is causing regressions (which I haven't investigated). Better to keep the code as is.

Reply:
ok, interesting.